From 8c7d584f6b945b3cf6b0246d6967b88d5cb94651 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 29 Oct 2016 17:41:22 +0100
Subject: [PATCH 23/47] Fix version comparison

MSVC reports:
'unary minus operator applied to unsigned type, result still unsigned'

.. so the version compare cannot work. Fix this with some casting.
---
 src/cpp/desktop/DesktopRVersion.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/desktop/DesktopRVersion.cpp b/src/cpp/desktop/DesktopRVersion.cpp
index 451cdee2fa..75fdb24ac5 100644
--- a/src/cpp/desktop/DesktopRVersion.cpp
+++ b/src/cpp/desktop/DesktopRVersion.cpp
@@ -586,7 +586,7 @@ int RVersion::compareTo(const RVersion& other) const
    int c;
 
    // First order by version, descending
-   c = -(version() - other.version());
+   c = -(static_cast<int>(version()) - static_cast<int>(other.version()));
    if (c != 0)
       return c;
 
-- 
2.17.0.windows.1

